app/services/order_creator.rb

```diff
@@ -0,0 +1,36 @@
+class OrderCreator
+  def self.call(cart, user, address_params)
+    new.call(cart, user, address_params)
+  end
+  def call(cart, user, address_params)
```

Comment:
Uma forma de garantir a consistencia dessa longa criação, é ultizar uma transaction, da uma olhada: 
https://medium.com/@kristenrogers.kr75/rails-transactions-the-complete-guide-7b5c00c604fc